src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.15;
+
+import "./ZkBobPool.sol";
+
+/**
+ * @title ZkBobCompoundingMixin
+ */
+abstract contract ZkBobCompoundingMixin is ZkBobPool {
+    using SafeERC20 for IERC20;
+
+    uint256 public investedAssetsAmount;
+    uint256 public investedSharesAmount;
+
+    struct YieldParams {
+        // ERC4626 vault address (or address(0) if not set)
+        address yield;
+        // maximum amount of underlying tokens that can be invested into vault
+        uint256 maxInvestedAmount;
+        // expected amount of underlying tokens to be left at the pool after successful rebalance
+        uint256 buffer;
+        // slippage/rounding protection buffer, small part of accumulated interest that is non-claimable
+        uint96 dust;
+        // address to receive accumulated interest during the rebalance
+        address interestReceiver;
+        // operator address (or address(0) if permissionless)
+        address yieldOperator;
+    }
```

Comment:
There are 2 possible snearios - we want to close the position in the vault or don't want.
It can be checked by comparing to variables - `totalInvestedAssets` and `previewWithdraw(totalInvestedShares)`
So, offchain bot can monitor these 2 variables and notify, when our tvl become less than users deposits. After that the team can decide, if position should be closed.
Insufficiency in tvl can be fullfilled by us just by transferring tokens.
So, in code function for closing position should be added, and possibility to decrease totalInvestedAmount manually by transferring tokens to pool